### ISSUE TITLE
Facebook\Authentication\AccessToken

### ISSUE THREAD

**Issue by gheesixz** [darkwhispering/facebook-sdk-codeigniter#16]
At first login, it was successful but when i try to refresh the page or click the Logout button, this is what i got..

Fatal error: Facebook::authenticate(): The script tried to execute a method or access a property of an incomplete object. Please ensure that the class definition &quot;Facebook\Authentication\AccessToken&quot; of the object you are trying to operate on was loaded _before_ unserialize() gets called or provide a __autoload() function to load the class definition in C:\*******\libraries\Facebook.php on line 213

on line 213 : 
if (!$access_token OR $access_token->isExpired())

i dont know what seems to be the problem here. 
im using xampp localhost..

**Comment by pawelkg** [comment]
I had the same problem. You need to register Facebook autoloader before loading session library.

**Comment by surajde16** [comment]
@pawelkg Can you explain in detail. Even i am facing the same problem

**Comment by pawelkg** [comment]
@surajde16 Just load FB autoloader before load library. I can't give you detailed instruction, because I used this library in heavily modified version of CI 1.5.4 mixed with CI 2.2.x

**Comment by cjogloc** [comment]
@pawelkg  @surajde16  i tried adding `require_once 'Facebook/autoload.php';` before class declaration in facebook library but it didn't work. Can you please tell me where to call the autoload file?

**Comment by darkwhispering** [comment]
@cjogloc @pawelkg @surajde16 @gheesixz 

You don't need to include the Facebook autoload manually. Please see my response to issue #23

**Comment by cjogloc** [comment]
@darkwhispering  composer autoload is already set to TRUE in config.php file. and i am already autoloading the facebook library in autoload.php still am getting the error.

**Comment by darkwhispering** [comment]
@cjogloc What version of the library are you using? And what version of CodeIgniter?

**Comment by cjogloc** [comment]
@darkwhispering i using codeigniter 3.0.6 and library version 3.0.0 beta 2
 from here https://github.com/darkwhispering/facebook-sdk-v4-codeigniter/tree/fb-sdk-v5-develop

**Comment by ongpang** [comment]
Hi bro, can you tell me please? I Logged in success but not access token.  Please help me bro!
------------------------------------------------------------
$user = $this->facebook->request('get', '/me?fields=id,name,email');
var_dump($user); exit;
------------------------------------------------------------
# Result
array (size=2)
  'error' => int 0
  'message' => string 'You must provide an access token.' (length=33)

**Comment by parbatpithiya** [comment]
Please include before of controller class and working like below :
<?php
defined('BASEPATH') OR exit('No direct script access allowed');
include(APPPATH."libraries/Facebook/src/autoload.php");

class Logiciellab extends CI_Controller {
---
}

**Comment by cjogloc** [comment]
@darkwhispering  @parbatpithiya  still not working.
If it is working for you. Can anyone give me the zip with folder structure (including SDK)?

**Comment by darkwhispering** [comment]
@cjogloc I have been unable to replicate the issue so far. I have however not had time to do any extensive testing so far and I also only the latest version of the FB SDK in the example controller I include in this library.

**Comment by code-freakz** [comment]
Hi, me too was facing  the same issue for the last two days. For me the problem was instead of doing " Install the Facebook PHP SDK by navigating to your applications folder in the terminal and run Composer with composer install. " i did run the composer in base directory. Please check if you have done the same mistake,

**Comment by darkwhispering** [comment]
Hi @code-freakz 

If you have the vendor folder outside of the application folder, then you are not using the CodeIgniter `composer_autoload` setting or you have configured a custom path for composer autoload in your config file.

The default location the composer_autoload looks for the composer autoload file is in `application/vendor/autoload.php` as stated in the config file

```php
/*
|--------------------------------------------------------------------------
| Composer auto-loading
|--------------------------------------------------------------------------
|
| Enabling this setting will tell CodeIgniter to look for a Composer
| package auto-loader script in application/vendor/autoload.php.
|
|	$config['composer_autoload'] = TRUE;
|
| Or if you have your vendor/ directory located somewhere else, you
| can opt to set a specific path as well:
|
|	$config['composer_autoload'] = '/path/to/vendor/autoload.php';
|
| For more information about Composer, please visit http://getcomposer.org/
|
| Note: This will NOT disable or override the CodeIgniter-specific
|	autoloading (application/config/autoload.php)
*/
$config['composer_autoload'] = TRUE;
```

The library loads fine for me using the default composer autoload and have the composer vendor folder in the application folder. I'm still trying to find the actual problem as it seems different solutions works for different people.

**Comment by darkwhispering** [comment]
Original issue posted by @gheesixz should be resolved in the last release, v3.0.0 now available in the master branch
